docs/language/README.md

```diff
@@ -1,35 +1,268 @@
+contain the string "widget".
+
+> NOTE: we should clarify keyword search vs substring match.
+
+As with the unix shell and legacy log search systems,
+the Zed language embraces a _pipeline_ model where a source of data
+is treated as a stream then one or more operators concatenated with
+the `|` symbol transform, filter, and aggregate the stream, e.g.,
+```
+widget | price > 1000
+```
+
+That said, the Zed language is declarative and
+the Zed compiler optimizes the data flow computation
+&mdash; e.g., implementing a Zed program often differently than
+the flow implied by the pipeline yet reaching the same result &mdash;
+much as a modern SQL engine optimizes a declarative SQL query.
+
+For example, the query from above is more efficiently implemented as
+a boolean AND operation instead of two pipeline stages,
+so the compiler is free to transform it to
+```
+widget and price > 1000
+```
+And since the "AND" syntax is optional (logical AND can be expressed as
+concatenation), this query can also be expressed as
+```
+widget price > 1000
+```
+
+To facilitate both a programming-like model as well as an ad hoc search
+experience, Zed has a canonical, long form that can be abbreviated
+using syntax that supports an agile interactive query workflow.
+For example, the canonical form of an aggregation uses the `summarize`
+keyword, as in
+```
+summarize count() by color
+```
+but this can be abbreviated by dropping the keyword whereby the compiler then
+uses the name of the aggregation function to resolve the ambiguity, e.g.,
+as in the shorter form
+```
+count() by color
+```
+Similarly, the canonical form of a search expression is a `filter` operator
+(and the "AND" operator is explicit in canonical form),
+so the example from above would be written canonically as
+```
+filter widget and price > 1000
+```
+Unlike typical log search systems, the Zed language operators are uniform:
+you can specify an operator including keyword search terms, boolean predicates,
+etc using the same syntax at any point in the pipeline.  For example,
+the predicate `count >= 10` can simply be tacked onto the output of a
+count aggregation using the filter from above and perhaps sorting
+the final about by `count` in a simple to type and edit fashion:
+```
+widget price > 1000 | count() by color | count >= 10 | sort count
+```
+The canonical form of this more complex query is:
+```
+filter widget and price > 1000
+| summarize count() by color
+| filter count >= 10
+| sort count
+```
+
+## SQL Compatiblity
+
+To encourage adoption by the vast audience of users who know and love SQL,
+a key goal of Zed is to support a superset of the data query language (DQL) portion
+of ANSI SQL.  For example, the above query can also be written in Zed as
+```
+SELECT count(), color
+WHERE widget AND price > 1000
+GROUP BY color
+HAVING count >= 10
+ORDER BY count
+```
+i.e., this SQL expression is a subset of the Zed language.
+Naturally, the SQL and Zed forms can be mixed and matched:
```

Comment:
```suggestion
Uniquely, the SQL and Zed forms can be mixed and matched:
```

Unless all the other languages mix SQL and non-SQL, it comes across to be more as a _unique_ differentiator rather than something that naturally follows. A _good_ differentiator, I think. 😉

Reply:
This doesn't read smoothly to me.  Perhaps "naturally" is too colloquial.  The intent here is "if you embrace this design philosophy of embedding SQL, then of course, you could mix and match Zed and SQL"
"Uniquely" as a parenthetical seems too obscure to me.  That requires more development or at least it should be the subject of a sentence, but it's not a point I was trying to make here.

Reply:
I'll just take it out